### PR TITLE
plugins: add fullscreen plugin

### PR DESCRIPTION
**Pull request by sanga**
I suppose at this point this is more of a request of interest (or lack of interest).

Tested manually and works. However should have a test added. Also it's somewhat unuseful at the moment (at least for me) as it won't work in an iframe which means that currently (I've filed a bug with the jupyter guys) it won't work in jupyter.

But, basically. Should I bother polishing this a bit more (adding a test etc)?

### CODE REVIEW

**Comment by ocefpaf** [comment]
@sanga this is definitely worth polishing! Nice addition.

File: folium/plugins/fullscreen.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+"""
+Fullscreen
+--------------
+
+https://github.com/brunob/leaflet.fullscreen
+
+Note: does not work from within an iframe, and because of this, does not
+work in jupyter notebook
+"""
```
Comment (ocefpaf) [inline folium/plugins/fullscreen.py]: I would put this in the class docstring. Add here just a quick description like `Adds fullscreen button to your maps.`

File: folium/plugins/fullscreen.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    def render(self, **kwargs):
+        super(Fullscreen, self).render()
+
+        figure = self.get_root()
+        assert isinstance(figure, Figure), ("You cannot render this Element "
+                                            "if it's not in a Figure.")
+
+        figure.header.add_children(
+            JavascriptLink(
+                "https://cdnjs.cloudflare.com/ajax/libs/"
```
Comment (ocefpaf) [inline folium/plugins/fullscreen.py]: It is fine to use long lines for URLs (and more readable/copy-n-pastable :wink:). Just add a `  # noqa` comment at the end to tell the linter to skip it.

**Comment by sanga** [comment]
Addressed all comments (I think). Did a little cleanup. Added (a rather soft) test. Added an example. And did a couple of 'drive by' fixes elsewhere. Let me know if you'd rather have the fixes in a separate branch. Oh and I force pushed with clean git history (let me know if you'd rather I didn't)

**Comment by sanga** [comment]
And force pushed a few cleanups to the fullscreen test

File: folium/plugins/fullscreen.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+from branca.element import MacroElement, Figure, JavascriptLink, CssLink
+
+
+class Fullscreen(MacroElement):
+    """
+    Adds a fullscreen button to your map.
+
+    Note: does not work from within an iframe, and because of this, does not
+    work in jupyter notebook
```
Comment (ocefpaf) [inline folium/plugins/fullscreen.py]: You mentioned you filed a bug about this, right? Can you add a cross-reference link here so people can follow the status of that?

File: tests/plugins/test_fullscreen.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+def test_fullscreen():
+    m = folium.Map([47, 3], zoom_start=1)
+    fs = plugins.Fullscreen()
+    m.add_children(fs)
+    m._repr_html_()
+
+    out = m._parent.render()
+
+    # verify that the fullscreen control was rendered
+    assert 'L.control.fullscreen().addTo' in out
```
Comment (ocefpaf) [inline tests/plugins/test_fullscreen.py]: :+1:

**Comment by ocefpaf** [comment]
LGTM. Only 1 minor comment.

@BibMartin if you find some time to take a quick look and, this is OK for you, go ahead and merge. Otherwise I will merge this next week once I get back in office.

**Comment by ocefpaf** [comment]
@sanga do you mind also adding an entry in the changelog file?

**Comment by sanga** [comment]
So I figured out who was rendering that iframe element (to which the allowfullscreen attr needed to be added). It wasn't jupyter actually. It's branca:

https://github.com/python-visualization/branca/pull/13

I've just noticed that I missed your comment about the changelog. Change coming shortly

**Comment by ocefpaf** [comment]
> So I figured out who was rendering that iframe element (to which the allowfullscreen attr needed to be added). It wasn't jupyter actually. It's branca:

Cool. Since that is an in-house issue let's get python-visualization/branca#13 sorted out first before we merge this. I am in a conference and I will take a look at this as soon as I get back. But it looks all good already. Thanks @sanga!

**Comment by sanga** [comment]
python-visualization/branca#13  is merged so this should be mergable now.

**Comment by ocefpaf** [comment]
Just used this plugin for a GIS lecture using `Jupyter` notebooks and `folium`. Thanks @sanga!

**Comment by sanga** [comment]
no worries! :)

**Pull request merged**